demos/blockfactory/workspacefactory/wfactory_controller.js

```diff
@@ -343,6 +343,10 @@ WorkspaceFactoryController.prototype.exportXmlFile = function(exportMode) {
     throw new Error(msg);
   }
 
+  // Unpack self-closing tags.  These tags fail when embedded in HTML.
+  // <block name="foo"/> -> <block name="foo"></block>
+  configXml = configXml.replace(/<(\w+)([^<]*)\/>/g, '<$1$2></$1>');
+
```

Comment:
A block name of "foo/>" would break this, but I checked and we already sanitize characters in the block factory.